x-pack/plugins/security_solution/server/endpoint/services/artifacts/manifest_manager/manifest_manager.ts

```diff
@@ -586,17 +665,30 @@ export class ManifestManager {
   private async listEndpointPolicies(page: number) {
     return this.packagePolicyService.list(this.savedObjectsClient, {
       page,
-      perPage: 20,
+      perPage: 1000,
       kuery: 'ingest-package-policies.package.name:endpoint',
     });
   }
 
-  private async listEndpointPolicyIds(page: number) {
-    return this.packagePolicyService.listIds(this.savedObjectsClient, {
-      page,
-      perPage: 20,
-      kuery: 'ingest-package-policies.package.name:endpoint',
-    });
+  private async listEndpointPolicyIds() {
```

Comment:
here you could use `iterateAllListItems` to not re-implement the paging. also, as we don't need the per-item callback for `iterateAllListItems`, it could receive a per-batch callback instead, so node can better optimize the merging of the batches - maybe it also helps with efficiency

Reply:
https://github.com/elastic/kibana/pull/160387/commits/dce677585777a01502768a857d113a2a25b1ce34